lib/redbird/plug/session/redis.ex

```diff
@@ -13,26 +15,30 @@ defmodule Plug.Session.REDIS do
     opts
   end
 
-  def get(_conn, namespaced_key, _init_options) do
-    case get(namespaced_key) do
-      :undefined -> {nil, %{}}
-      value -> {namespaced_key, value |> :erlang.binary_to_term()}
+  def get(conn, prospective_key, _init_options) do
+    with {:ok, key, _} <- Key.extract_key(prospective_key),
+         {:ok, _verified_key} <- Key.verify(key, conn),
+         value when is_binary(value) <- get(prospective_key) do
+      {prospective_key, Crypto.safe_binary_to_term(value)}
+    else
+      _ -> {nil, %{}}
     end
   end
 
   def put(conn, nil, data, init_options) do
-    put(conn, add_namespace(generate_random_key()), data, init_options)
+    put(conn, Key.generate(), data, init_options)
   end
 
-  def put(_conn, namespaced_key, data, init_options) do
-    value = :erlang.term_to_binary(data)
+  def put(conn, key, data, init_options) do
+    key
+    |> Key.sign_key(conn)
+    |> set_key_with_retries(:erlang.term_to_binary(data), session_expiration(init_options), 1)
```

Comment:
What do you think about moving the `Crypto.safe_binary_to_term(value)` in line 22 and this `:erlang.term_to_binary(data)` to private functions in this module that are called `encode/1` and `decode/1`? 

I know I suggested they move to the `Key` module, but you correctly pointed out they're not the keys. They're the values. But I still think that extracting them somewhere would make it easier to see the parity between encoding/decoding. 

It's possible I'm reading that wrong, and those aren't opposite operations. If that's the case, feel free to ignore this. But otherwise, what do you think? It might just give a little more clarity as to where we do the encoding/decoding of the session data?